cmd/install.go

```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
-				return "", err
-			}
-			if err = utils.CreateFolder(ctx, nodeInstallPath, config.Force); err != nil {
-				return "", err
-			}
-		} else {
-			return "", err
-		}
+	if err = CreateUtilityConfigFile(ctx, config); err != nil {
+		log.WithContext(ctx).WithError(err).Error("Failed to create pastel-utility config file")
+		return err
 	}
 
-	return "", nil
-}
-
-func runComponentsInstall(ctx context.Context, config *configs.Config, installCommand constants.ToolType) (err error) {
-	if err = InitializeFunc(ctx, config); err != nil {
+	if err = createInstallDir(ctx, config, config.PastelExecDir); err != nil {
+		//error was logged inside createInstallDir
 		return err
 	}
 
-	if _, err = initNodeDownloadPath(ctx, config, config.PastelExecDir); err != nil {
+	if err = checkInstalledPackages(ctx); err != nil {
+		log.WithContext(ctx).WithError(err).Error("Missing packages...")
 		return err
 	}
 
-	switch installCommand {
-	case constants.PastelD:
+	if installCommand == constants.PastelD ||
+		installCommand == constants.WalletNode ||
+		installCommand == constants.SuperNode {
 		if err = installComponent(ctx, config, constants.PastelD, config.Version); err != nil {
 			return err
 		}
-	case constants.WalletNode:
-		if err = installComponent(ctx, config, constants.PastelD, config.Version); err != nil {
-			return err
-		}
-
-		if err = installComponent(ctx, config, constants.WalletNode, config.Version); err != nil {
-			return err
-		}
-
+	}
+	if installCommand == constants.WalletNode ||
+		installCommand == constants.SuperNode {
 		if err = installComponent(ctx, config, constants.RQService, config.Version); err != nil {
 			return err
 		}
-	case constants.SuperNode:
-		if err = installComponent(ctx, config, constants.PastelD, config.Version); err != nil {
+	}
+	if installCommand == constants.WalletNode {
+		if err = installComponent(ctx, config, constants.WalletNode, config.Version); err != nil {
 			return err
 		}
-
+	}
+	if installCommand == constants.SuperNode {
 		if err = installComponent(ctx, config, constants.SuperNode, config.Version); err != nil {
 			return err
 		}
-
-		if err = installComponent(ctx, config, constants.RQService, config.Version); err != nil {
-			return err
-		}
-
 		// Open ports
 		openErr := openPort(ctx, constants.PortList)
 		if openErr != nil {
 			return openErr
 		}
 
-		log.WithContext(ctx).Info("Installing dd-service...")
 		if err = installDupeDetection(ctx, config); err != nil {
-			log.WithContext(ctx).Error("Installing dd-service executable failed")
+			log.WithContext(ctx).WithError(err).Error("Installing dd-service executable failed")
 			return err
 		}
-		log.WithContext(ctx).Info("The dd-service Installed Successfully")
 	}
 
 	return nil
 }
 
+func createInstallDir(ctx context.Context, config *configs.Config, installPath string) error {
+	defer log.WithContext(ctx).Infof("Install path is %s", installPath)
+
+	if err := utils.CreateFolder(ctx, installPath, config.Force); os.IsExist(err) {
```

Comment:
missing the case utils.CreateFolder() failed and error is not satisfied os.IsExist(err).